wagtail/admin/templates/wagtailadmin/shared/button.html

```diff
@@ -1,7 +1,17 @@
 {% load wagtailadmin_tags %}
-<a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>
-    {% if button.icon_name %}
-        {% icon name=button.icon_name %}
-    {% endif %}
-    {{ button.label }}
-</a>
+
+{% if button.url %}
+    <a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>
+        {% if button.icon_name %}
```

Comment:
Maybe we should use `{% fragment` here, so we do not have to repeat this block of logic?

```
{% fragment as content %}
    {% if button.icon_name %}
        {% icon name=button.icon_name %}
    {% endif %}
    {{ button.label }}
{% endfragment %}
```

Then, we can use it in the inside like this.

```
<a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>{{ content }}</a>
```